scan/spec/test_command_generator_spec.rb

```diff
@@ -166,6 +198,32 @@
       end
     end
 
+    it "uses a device without version specifier" do
+      options = { project: "./examples/standard/app.xcodeproj", device: "iPhone 6s" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+
+      result = Scan::TestCommandGenerator.generate
+      expect(result).to start_with([
+                                     "set -o pipefail &&",
+                                     "env NSUnbufferedIO=YES xcodebuild",
+                                     "-scheme app",
+                                     "-project ./examples/standard/app.xcodeproj",
+                                     # expect the single highest versioned iOS simulator device available with matching name
+                                     "-destination 'platform=iOS Simulator,id=021A465B-A294-4D9E-AD07-6BDC8E186343'",
+                                     "-derivedDataPath '#{Scan.config[:derived_data_path]}'",
+                                     :build,
+                                     :test
+                                   ])
+    end
+
+    it "rejects devices with versions below deployment target" do
+      options = { project: "./examples/standard/app.xcodeproj", device: "iPhone 5 (8.4)" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+
+      result = Scan::TestCommandGenerator.generate
+      # FIXME: expect UI error starting "No simulators found that are equal to the version of specifier"
```

Comment:
Ah! Makes sense... So I think the best way to handle that would be to call `UI.crash!(<message>)` instead of `UI.error` at the place of the link you shared, because if there is no simulator found, we can't proceed anyway right? 

That would also make this test easy to implement with something like this in `rspec`: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher